common/primitives/src/parquet_model.rs

```diff
@@ -0,0 +1,57 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use scale_info::Type;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_std::prelude::*;
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct ParquetModel {
+  _type: ParquetType,
+  compression: CompressionCodec,
+  statistics: bool,
+  bloom_filter: bool,
+  optional: bool
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+pub enum ParquetType {
```

Comment:
@dopatraman I think I like this. Let's keep going down this path. Two complications to work through on ParquetTypes around [Logical Types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md):

1. Some are easy. DSNP needs string types for example, but those fit into the enum pattern
    - We also use numerical types and those take parameters so an enum might not work. Not sure how we would want to handle those, but they are fairly simple still.
2. We don't use the Nested Types in DSNP, but I am very unsure about how we could model those
    - Likely we should not support those v1, but might be worth considering.

Reply:
re: types that take parameters, I believe Rust enums can be modeled with params: https://doc.rust-lang.org/rust-by-example/custom_types/enum.html

re: nested types, good point. I will tinker a bit and see what we can do.